docs/ref/models/querysets.txt

```diff
@@ -1628,6 +1628,12 @@ specify the related objects you want to lock in ``select_for_update(of=(...))``
 using the same fields syntax as :meth:`select_related`. Use the value ``'self'``
 to refer to the queryset's model.
 
+It is not possible to use ``select_for_update()`` when selecting nullable
+related objects. If you know that the related objects are not null, you can
+overcome this problem by excluding null values. For example::
+
+    entries = Person.objects.select_for_update().exclude(hometown=None)
```

Comment:
Is this example correct? It doesn't given an error without the exclude which I thought is what would be demonstrated.

Reply:
You're right, it was missing the `select_related` call that makes it necessary.